### PR TITLE
fix(synthetic-shadow): querySelectors not returning

### PR DESCRIPTION
**Pull request by jodarove**
manually inserted nodes in same shadow tree

## Details
This PR makes `querySelector`, `querySelectorAll`, `getElementsBy*` to include manually inserted elements in the same shadow of the query context.

## Does this PR introduce breaking changes?

* ✅ `No, it does not introduce breaking changes.`

### CODE REVIEW

File: packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts [hunk 1]
```diff
@@ -330,11 +342,21 @@ function getFilteredArrayOfNodes<T extends Node>(
...
-        if (!isUndefined(ownerKey) || shadowDomSemantic === ShadowDomSemantic.Enabled) {
-            // The patch is enabled or `context` is an element rendered by lwc
-            filtered = ArrayFilter.call(unfilteredNodes, elm => getNodeOwnerKey(elm) === ownerKey);
+        if (!isUndefined(ownerKey)) {
+            // context is handled by lwc, using getNodeNearestOwnerKey to include manually inserted elements in the same shadow.
+            filtered = ArrayFilter.call(
+                unfilteredNodes,
+                elm => getNodeNearestOwnerKey(elm) === ownerKey
+            );
+        } else if (shadowDomSemantic === ShadowDomSemantic.Enabled) {
```
Comment (ravijayaramappa) [inline packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts]: May be in a separate PR, the enum values should be in uppercase

File: packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts [hunk 2]
```diff
@@ -303,8 +316,7 @@ function querySelectorPatched(this: Element /*, selector: string*/): Element | n
             );
             return isUndefined(elm) ? null : elm;
         } else {
-            const elm = nodeList[0];
-            return isUndefined(elm) ? null : elm;
+            return nodeList.length === 0 ? null : nodeList[0];
```
Comment (ekashida) [inline packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts]: Am I missing something or does this mean the same thing as before?

Comment (jodarove) [reply packages/@lwc/synthetic-shadow/src/faux-shadow/element.ts]: the same thing, @ravijayaramappa says this way reveals more intent.
